pkg/router/controller/shared_secret_manager.go

```
@@ -0,0 +1,290 @@
+			}
+		}
+	}
+
+	m.registeredRoutes[key] = referencedSecret{
+		secretName: secretName,
+		handler:    handler,
+		restricted: restricted,
+	}
+
+	infKey := m.getInformerKey(namespace, secretName, restricted)
+	infState, exists := m.informers[infKey]
```

golangci-lint (2.12.2)

[error] ineffectual assignment to infState

(ineffassign)